Gradle 5.0-milestone-1 throws an exception when accessing properties in settings.gradle

<!--- 
Please follow the instructions below. We receive dozens of issues every week, so to stay productive, we will close issues that don't provide enough information. 

Please open Android-related issues on the Android Issue Tracker at https://source.android.com/source/report-bugs
Please open Kotlin DSL-related issues at https://github.com/gradle/kotlin-dsl/issues
Please open Gradle Native-related issues at https://github.com/gradle/gradle-native/issues
-->

<!--- Provide a brief summary of the issue in the title above -->

### Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
This statement in the settings.gradle should not cause gradle to immediately throw an exception
```
if (properties.someProjectProperty != 'true') {
    include 'subproject'
}
```
"someProjectProperty" is a project property that is set to true in the root gradle.properties
### Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->
```
FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/ericp/Projects/someproject/settings.gradle' line: 67

* What went wrong:
A problem occurred evaluating settings 'someproject'.
> org.gradle.initialization.DefaultSettings_Decorated.getConvention()Lorg/gradle/api/plugins/Convention;

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s

```
### Steps to Reproduce (for bugs)
Add a conditional statement to settings.gradle that uses a project property

### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- A build scan `https://scans.gradle.com/get-started` is ideal -->
Gradle 5.0-milestone-1

--- a/subprojects/core/src/integTest/groovy/org/gradle/api/SettingsDslIntegrationSpec.groovy
+++ b/subprojects/core/src/integTest/groovy/org/gradle/api/SettingsDslIntegrationSpec.groovy
@@ -33,6 +33,23 @@ class SettingsDslIntegrationSpec extends AbstractIntegrationSpec {
         succeeds('help')
     }
 
+    def "can dynamically access properties"() {
+
+        given:
+        file("gradle.properties") << "someProjectProperty=true"
+        settingsFile << """
+            if (properties.someProjectProperty == 'true') {
+                println('signal')
+            }
+        """
+
+        when:
+        succeeds('help')
+
+        then:
+        outputContains('signal')
+    }
+
     def "Can type-safely use ExtensionAware with the Groovy DSL"() {
         when:
         def answerFile = "answerHolder.gradle"

--- a/subprojects/core/src/main/java/org/gradle/initialization/DefaultSettings.java
+++ b/subprojects/core/src/main/java/org/gradle/initialization/DefaultSettings.java
@@ -27,6 +27,7 @@ import org.gradle.api.internal.ExtensibleDynamicObject;
 import org.gradle.api.internal.FeaturePreviews;
 import org.gradle.api.internal.FeaturePreviews.Feature;
 import org.gradle.api.internal.GradleInternal;
+import org.gradle.api.internal.HasConvention;
 import org.gradle.api.internal.SettingsInternal;
 import org.gradle.api.internal.file.FileResolver;
 import org.gradle.api.internal.initialization.ClassLoaderScope;
@@ -35,6 +36,7 @@ import org.gradle.api.internal.plugins.DefaultObjectConfigurationAction;
 import org.gradle.api.internal.plugins.PluginManagerInternal;
 import org.gradle.api.internal.project.AbstractPluginAware;
 import org.gradle.api.internal.project.ProjectRegistry;
+import org.gradle.api.plugins.Convention;
 import org.gradle.api.plugins.ExtensionContainer;
 import org.gradle.caching.configuration.BuildCacheConfiguration;
 import org.gradle.configuration.ScriptPluginFactory;
@@ -54,7 +56,7 @@ import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
 
-public class DefaultSettings extends AbstractPluginAware implements SettingsInternal, DynamicObjectAware {
+public class DefaultSettings extends AbstractPluginAware implements SettingsInternal, DynamicObjectAware, HasConvention {
     public static final String DEFAULT_BUILD_SRC_DIR = "buildSrc";
     private ScriptSource settingsScript;
 
@@ -336,6 +338,11 @@ public class DefaultSettings extends AbstractPluginAware implements SettingsInte
 
     @Override
     public ExtensionContainer getExtensions() {
+        return extensibleDynamicObject.getConvention();
+    }
+
+    @Override
+    public Convention getConvention() {
         return extensibleDynamicObject.getConvention();
     }
 }

